optuna/storages/_rdb/storage.py

```diff
@@ -168,12 +194,9 @@ def create_new_study(self, study_name: Optional[str] = None) -> int:
 
     def delete_study(self, study_id: int) -> None:
 
-        session = self.scoped_session()
-
-        study = models.StudyModel.find_or_raise_by_id(study_id, session)
-        session.delete(study)
-
-        self._commit_with_integrity_check(session)
+        with _scoped_session(self.scoped_session) as session:
+            study = models.StudyModel.find_or_raise_by_id(study_id, session)
+            session.delete(study)
```

Comment:
We may need to try and catch/silence the `IntegrityError` to keep the behavior consistent.

Reply:
Ditto for most occasions where we used to call `self._commit_with_integrity_check`.